File: lessons/compiled_lang/java/src/o00.flags.java
import java.io.PrintStream;

// Not public - see a00.output.java for why.
class O00Flags {
    private static void usage(PrintStream out, String scriptName) {
        out.println();
        out.println("Usage: " + scriptName + " [-c|-e|-l|-k|-p|-m|-t] [-h|-?]");
        out.println();
        out.println("  -c  Coffee");
        out.println("  -e  Espresso");
        out.println("  -l  Latte");
        out.println("  -k  Machiato");
        out.println("  -p  Capucino");
        out.println("  -m  Mocha");
        out.println("  -t  Tea");
        out.println("  -h  Display this help message");
        out.println("  -?  Display this help message");
        out.println();
    }

    public static void main(String[] args) {
        // the wrapper script's own Makefile-injected "invoked.as" system
        //  property stands in for argv[0] here (see ../Makefile).
        String scriptName = System.getProperty("invoked.as");

        if (args.length == 0) {
            usage(System.err, scriptName);
            System.exit(1);
            return;
        }

        switch (args[0]) {
            case "-c":
                System.out.println("You ordered a Coffee.");
                break;
            case "-e":
                System.out.println("You ordered an Espresso.");
                break;
            case "-l":
                System.out.println("You ordered a Latte.");
                break;
            case "-k":
                System.out.println("You ordered a Machiato.");
                break;
            case "-p":
                System.out.println("You ordered a Capucino.");
                break;
            case "-m":
                System.out.println("You ordered a Mocha.");
                break;
            case "-t":
                System.out.println("You ordered a Tea.");
                break;
            case "-h":
            case "-?":
                usage(System.out, scriptName);
                break;
            default:
                usage(System.err, scriptName);
                System.exit(1);
        }
    }
}

File: lessons/compiled_lang/cs/src/o00.flags.cs
using System;

class O00Flags
{
    static void Usage(System.IO.TextWriter outp, string scriptName)
    {
        outp.WriteLine();
        outp.WriteLine("Usage: " + scriptName + " [-c|-e|-l|-k|-p|-m|-t] [-h|-?]");
        outp.WriteLine();
        outp.WriteLine("  -c  Coffee");
        outp.WriteLine("  -e  Espresso");
        outp.WriteLine("  -l  Latte");
        outp.WriteLine("  -k  Machiato");
        outp.WriteLine("  -p  Capucino");
        outp.WriteLine("  -m  Mocha");
        outp.WriteLine("  -t  Tea");
        outp.WriteLine("  -h  Display this help message");
        outp.WriteLine("  -?  Display this help message");
        outp.WriteLine();
    }

    static void Main(string[] args)
    {
        string scriptName = Environment.GetCommandLineArgs()[0];

        if (args.Length == 0)
        {
            Usage(Console.Error, scriptName);
            Environment.Exit(1);
            return;
        }

        switch (args[0])
        {
            case "-c":
                Console.WriteLine("You ordered a Coffee.");
                break;
            case "-e":
                Console.WriteLine("You ordered an Espresso.");
                break;
            case "-l":
                Console.WriteLine("You ordered a Latte.");
                break;
            case "-k":
                Console.WriteLine("You ordered a Machiato.");
                break;
            case "-p":
                Console.WriteLine("You ordered a Capucino.");
                break;
            case "-m":
                Console.WriteLine("You ordered a Mocha.");
                break;
            case "-t":
                Console.WriteLine("You ordered a Tea.");
                break;
            case "-h":
            case "-?":
                Usage(Console.Out, scriptName);
                break;
            default:
                Usage(Console.Error, scriptName);
                Environment.Exit(1);
                break;
        }
    }
}
